<a href="https://colab.research.google.com/github/Nayab189/flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*


I chose the **Refresh / Content Opportunity Scoring** lane because it focuses on helping editors prioritize which content pages should be reviewed first. The starter dataset contains observable search and engagement signals such as impressions, clicks, CTR, sessions, freshness, and content age, making it suitable for this task. The goal is to support editorial decisions by ranking pages for review rather than replacing human judgment.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research Question:** Which content pages should be reviewed first for refresh based on observable search and engagement signals?

- **Decision:** Decide which pages should be reviewed first.
- **Who acts?** Content editors or the SEO team.
- **Unit of analysis:** One content page.
- **Output:** A ranked refresh priority score.
- **Action:** Refresh, expand, monitor, or postpone a page.
- **Cost of a wrong recommendation:** A false positive wastes editor time on a page that does not need attention. A false negative may miss an important declining page, leading to further loss of visibility or traffic.
- **Why data/ML?** This decision depends on multiple observable signals that are difficult to combine consistently using simple manual rules alone.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

print("Dataset shape:", df.shape)
print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nAverage impressions:")
print(df["impressions_90d"].mean())

print("\nAverage sessions:")
print(df["sessions_90d"].mean())

Dataset shape: (30000, 44)

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average impressions:
5200.3663

Average sessions:
37.066633333333336


The starter dataset contains **30,000 content pages** with **44 columns**, providing enough examples for exploratory analysis.

The distribution of `trend_direction` shows that **16,262 pages are currently marked as "down"**, indicating that many pages may require review, while other pages are classified as stable, up, new, or flat.

The average **90-day impressions are approximately 5,200**, and the average **90-day sessions are about 37**, showing that many pages have measurable search visibility and user engagement.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This provides decision-support by ranking pages for review using observed historical search and engagement signals. The results are observational and directional. They can help prioritize human review but do not prove that refreshing a page will improve traffic or rankings. This work does not predict Google's algorithm or establish a causal relationship between content changes and future performance.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.